### PR TITLE
Fix findAllRefs, getHighlightSpans, renameLocs, renameInfo for default exports and functions expressions

### LINKED ISSUE #3065: Can't Refactor -> Rename a default export name in Visual Studio
Hi. I'm using VS 2013 and TypeScript 1.5 Beta. Amazing work, as always.

I can't seem to rename default exports via "Rename". Consider this:

```TypeScript
export default class InfoPanelViewModel {
}
```

- Invoke the Rename operation 
- Punch in a new name, submit via click or return
- Notice no change in the symbol's name

Doing a Refactor -> Rename does nothing. Tested with `function`, seems to be for anything `export default`. Removing the `default` keyword allows me to rename the symbol. Adding it back prevents it again.

Is this by design, or a bug? If it's by design, perhaps a warning or error would be useful.

### LINKED ISSUE #3230: Named function expressions get incorrect display names when renaming
image

Note that the dialog box shows "Rename '__function'" instead of "Rename 'foo'"

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
A follow up from #3247, but merges into the `master` branch.

* Fixes #3065
* Fixes #3230

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
@mhegazy I know we discussed in the other issue that we wanted to move the `isImportOrExportSpecifierName` check to `getSymbolInfo` but do you think we can merge this change in in the interest of keeping the PR small?

**Comment by DanielRosenwasser** [comment]
Ping @yuit @mhegazy @vladima

**Comment by yuit** [comment]
Can you added rename cases for function expression and class expression?
```typescript
// [|foo|] -> this should not get updated
function foo () {}

// [|foo|]  -> this should get updated.  this is a kind of a grey area of what we should do
var x = function [|foo|] () {}  // rename foo here should not affect foo outside

var y = function () {
   [|foo|](); // Rename foo here 
}
```

```typescript
// [|foo|] -> this should not get updated
class Foo () {}

// [|foo|]  -> this should get updated.  this is a kind of a grey area of what we should do
var x = class [|Foo|] () { \\ Rename Foo here
    getClassName() {
        [|Foo|] -> rename Foo here
    }
} 

var y = class {
   getSomeName() {
      [|Foo|].  -> rename Foo here
   }
}
```
Similar for findAllRefs

File: src/services/services.ts [hunk 1]
```diff
@@ -4965,8 +4965,8 @@ namespace ts {
             // Compute the meaning from the location and the symbol it references
             let searchMeaning = getIntersectingMeaningFromDeclarations(getMeaningFromLocation(node), declarations);
 
-            // Get the text to search for, we need to normalize it as external module names will have quote
-            let declaredName = getDeclaredName(symbol, node);
+            // Get the text to search for, we need to normalize it as external module names will have quotes
+            let declaredName = getDeclaredName(typeChecker, symbol, node);
```
Comment (yuit) [inline src/services/services.ts]: What you mean by normalization? stripping quotes in ```getDeclaredName```? if so could you mention this in the comment

File: src/services/services.ts [hunk 2]
```diff
@@ -5013,76 +5013,41 @@ namespace ts {
...
-                // to a symbol with the name "__function".
-                let name: string;
-                if (functionExpression && functionExpression.name) {
-                    name = functionExpression.name.text;
-                }
-
-                // If this is an export or import specifier it could have been renamed using the as syntax.
-                // if so we want to search for whatever under the cursor, the symbol is pointing to the alias (name)
-                // so check for the propertyName.
+            function getInternedName(symbol: Symbol, location: Node, declarations: Declaration[]) {
```
Comment (yuit) [inline src/services/services.ts]: keep return type

File: src/services/utilities.ts [hunk 3]
```diff
@@ -652,4 +652,34 @@ namespace ts {
             typechecker.getSymbolDisplayBuilder().buildSignatureDisplay(signature, writer, enclosingDeclaration, flags);
         });
     }
+
+    export function getDeclaredName(typeChecker: TypeChecker, symbol: Symbol, location: Node) {
```
Comment (yuit) [inline src/services/utilities.ts]: Add return type

**Pull request merged**